src/core/probe/kernel/kprobe.rs

```diff
@@ -39,19 +41,25 @@ impl ProbeBuilder for KprobeBuilder {
             bail!("Kprobe builder already initialized");
         }
 
+        register_filter_handler(
```

Comment:
this is obviously not the right place to do this (this placeholder has to be moved somewhere else), unless we want to add some kind of lazy registration mechanism that registers once.
Same for the other program types.

Reply:
> this is not the right place to do this (this placeholder has to be moved somewhere else), unless we want to add some kind of lazy registration mechanism that registers once (see kretprobe case).

erhm, this is actually something we might want in the case kprobes "matching" our kretprobes would consume filtering.
This is not the case, ATM. Furthermore, the series renames the function and section so we don't have that overlap.

Reply:
> It's probably OK to do this call in the filtering part or probe manager directly. It's only loosely linked to those probe builder anyway.

On second thought .init() was not that bad, but the doubt was between that and the manager. I followed your suggestion and moved this specific part to the manager.